### PR TITLE
update dependencies

### PR DESCRIPTION
**Pull request by heron2014**
### Ready

This PR updates the dependencies from year ago . restructure the project, convert es5 into es6 and update testing
@nofootnotes could you do the honour and check this PR? :smile:

### CODE REVIEW

**Comment by claireinez** [comment]
So is this ready or not? I'm confused!

File: test/socket_test.js [hunk 1]
```diff
@@ -1,121 +1,121 @@
...
-  var client = io.connect(socketURL, ioptions);
-  client.on("connect", function(data) {
-    client.emit("message");
-    t.ok(true, "App did not crash on empty message");
-    client.disconnect();
-    t.end();
-  });
-});
\ No newline at end of file
+// var io = require('socket.io-client'),
```
Comment (claireinez) [inline test/socket_test.js]: Maybe just delete the file, rather than commenting it all out? :smile:

**Comment by heron2014** [comment]
@nofootnotes not ready, have you ever done coverage for error handlers? 
I am missing two lines! so annoying: 

coverage

File: test/socket_test.js (hunk 1 above)
Comment (heron2014) [inline test/socket_test.js]: I have deleted this one! :+1:

**Comment by heron2014** [comment]
I have added one more test but it didn't make much difference, still missing these lines to have 100% coverage! But I run out of ideas.

**Comment by claireinez** [comment]
No... My testing is pretty poor these days so not sure how to go about that!

**Comment by heron2014** [comment]
Cool! We figure out on the way.. :) If I come up with good test I will create other PR for this enhancement!  Feel free to merge it whenever you have time :)

**Comment by heron2014** [comment]
@nofootnotes our README is so ambitious haha I was today struggling with one test for at least 2h! But its again 100% yay I can go to pub now :)

**Comment by claireinez** [comment]
👍

**Pull request merged**